File: integrations/src/main/java/framework/core/data/IntegrationResponse.java
package framework.core.data;

import org.springframework.http.HttpStatus;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class IntegrationResponse {
    private IntegrationResponseStatus responseStatus;
    private HttpStatus httpStatus;
    private String responseBody;
    private String errorMessage;

    public boolean isSuccess() {
        return IntegrationResponseStatus.SUCCESS == responseStatus;
    }

    public static IntegrationResponse successResponse(HttpStatus httpStatus, String responseBody) {
        return new IntegrationResponse(IntegrationResponseStatus.SUCCESS, httpStatus, responseBody, null);
    }

    public static IntegrationResponse serverErrorResponse(HttpStatus httpStatus, String responseBody) {
        return new IntegrationResponse(IntegrationResponseStatus.SERVER_ERROR, httpStatus, responseBody, null);
    }

    public static IntegrationResponse clientErrorResponse(HttpStatus httpStatus, String responseBody) {
        return new IntegrationResponse(IntegrationResponseStatus.CLIENT_ERROR, httpStatus, responseBody, null);
    }

    public static IntegrationResponse serverErrorResponse(String error) {
        return new IntegrationResponse(IntegrationResponseStatus.SERVER_ERROR, null, null, error);
    }

    public static IntegrationResponse clientErrorResponse(String error) {
        return new IntegrationResponse(IntegrationResponseStatus.CLIENT_ERROR, null, null, error);
    }
}

File: integrations/src/main/kotlin/framework/core/data/IntegrationResponse.kt
package framework.core.data

import org.springframework.http.HttpStatus

data class IntegrationResponse(
    val responseStatus: IntegrationResponseStatus,
    val httpStatus: HttpStatus?,
    val responseBody: String?,
    val errorMessage: String?
) {
    fun isSuccess(): Boolean {
        return IntegrationResponseStatus.SUCCESS == responseStatus
    }

    companion object {
        @JvmStatic
        fun successResponse(httpStatus: HttpStatus, responseBody: String?): IntegrationResponse {
            return IntegrationResponse(IntegrationResponseStatus.SUCCESS, httpStatus, responseBody, null)
        }

        @JvmStatic
        fun serverErrorResponse(httpStatus: HttpStatus, responseBody: String?): IntegrationResponse {
            return IntegrationResponse(IntegrationResponseStatus.SERVER_ERROR, httpStatus, responseBody, null)
        }

        @JvmStatic
        fun clientErrorResponse(httpStatus: HttpStatus, responseBody: String?): IntegrationResponse {
            return IntegrationResponse(IntegrationResponseStatus.CLIENT_ERROR, httpStatus, responseBody, null)
        }

        @JvmStatic
        fun serverErrorResponse(error: String?): IntegrationResponse {
            return IntegrationResponse(IntegrationResponseStatus.SERVER_ERROR, null, null, error)
        }

        @JvmStatic
        fun clientErrorResponse(error: String?): IntegrationResponse {
            return IntegrationResponse(IntegrationResponseStatus.CLIENT_ERROR, null, null, error)
        }
    }
}
